modules/brainbrowser/ajax/minc.php

```diff
@@ -22,13 +22,56 @@
 
 $headers = array();
 
-$query     = "select File from files where FileID = :MincID";
-$minc_file = $DB->pselectOne($query, array('MincID' => $_REQUEST['minc_id']));
-$minc_file = getMincLocation() . $minc_file;
+if (strpos(escapeshellcmd($_REQUEST['minc_id']), 'l') !== false) {
+    list($l, $id) = explode('l', escapeshellcmd($_REQUEST['minc_id']));
 
+    switch ($l) {
+    case 1:
+        $query = "SELECT minc_location FROM mri_protocol_violated_scans " .
+        "WHERE ID = :LogID";
+        break;
+    case 2:
+        $query = "SELECT MincFile FROM mri_violations_log WHERE LogID = :LogID";
+        break;
+    case 3:
+        $query = "SELECT MincFile FROM MRICandidateErrors WHERE ID = :LogID";
+        break;
+    default:
+        header("HTTP/1.1 400 Bad Request");
+        exit();
+    }
+
+    if (!empty($query)) {
+        $minc_file = $DB->pselectOne($query, array('LogID' => $id));
+        $file      = implode('/', array_slice(explode('/', $minc_file), -2));
 
-$header      = $_REQUEST['minc_headers'];
-$header_data = $_REQUEST['raw_data'];
+        if (strpos($minc_file, 'assembly') !== false) {
+            if (strpos($minc_file, 'assembly') === 0) {
+                $minc_file = getMincLocation() . $minc_file;
+            } else {
+                $minc_file = $minc_file;
+            }
+        } else {
+            $minc_file = getMincLocation() . "trashbin/" . $file;
+        }
+    }
+} else {
+    $query     = "select File from files where FileID = :MincID";
+    $minc_file = $DB->pselectOne(
+        $query,
+        array(
+         'MincID' => escapeshellcmd($_REQUEST['minc_id']),
```

Comment:
I don't see what's gained by calling `escapeshellcmd` here. It's being passed to a prepared statement binding, not being passed to a shell.